lib/pbench/server/database/models/datasets.py

```diff
@@ -1054,20 +1081,55 @@ def setvalue(
         "a.b.c" a value of "bar", and then query "a", you'll get back
         "a": {"b": {"c": "bar"}}}
 
+        We handle two keys specially:
+
+            dataset.name:   This is the primary resource name of a dataset, and
```

Comment:
Can we consider making sure the original name is always saved so that it can be restored via the UI?

Reply:
I considered that, decided not to waste extra space because it can be reconstructed from `dataset.metalog.{script + config + date}` just as it was created in the first place.

I also considered that we could add a `--name` option on the wrapper scripts on the agent side; but that would have to be communicated to the server via a new `metadata.log` value (e.g., `pbench/name`) so it'd still be recorded immutably (and available to clients as metadata).

Reply:
That presumes we have control over the behavior of the agent, which may not be the case.  I think we should preserve the original values either via an explicit field, or via mechanism where we can re-read from the original `metadata.log`.

Reply:
Tracking this in a Jira issue as future work.

Reply:
I really don't understand your context here; if you really want to track some possible future work, the Jira issue needs a lot more information about your concern.

Are you trying to say that a tarball night not have a `metadata.log` with a `pbench` section containing `script`, `config` and `date` values? There's also already a `name` property, recording the original agent-side name. If we're supposed to operate without any of the `metadata.log` properties, then we need a fair amount of changes in the intake processing (e.g., the creation date comes from here, and we use it in indexing).

Or do you have something else in mind entirely that I'm not grasping?

Reply:
What I mean is that ...

> it can be reconstructed from `dataset.metalog.{script + config + date}` just as it was created in the first place

... implies that every pbench agent does it that way.  The server can't assume anything about the value of name since it does not have "control" over how it is set.  The value is controlled by the Pbench Agent code.

"We" as developers "control" the behavior, but enforcing that behavior will never be possible on the server side, unless the server ignores `name` and creates it from those fields.

So I'd rather see the server offer a way to either "save" the original name value in an immutable field, or offer a mechanism to restore it from the original `metadata.log` file.